x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/GenerateInitialBuiltinUsersPasswordListener.java

```diff
@@ -66,32 +73,25 @@ public void accept(SecurityIndexManager.State previousState, SecurityIndexManage
         }
     }
 
-    private void outputOnSuccess(SecureString elasticPassword, SecureString kibanaSystemPassword) {
-        LOGGER.info("");
-        LOGGER.info("-----------------------------------------------------------------");
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("Password for the elastic user is: " + elasticPassword);
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("Password for the kibana_system user is: " + kibanaSystemPassword);
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("Please note these down as they will not be shown again.");
-        LOGGER.info("");
-        LOGGER.info("You can use 'bin/elasticsearch-reset-elastic-password' at any time");
-        LOGGER.info("in order to reset the password for the elastic user.");
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("You can use 'bin/elasticsearch-reset-kibana-system-password' at any time");
-        LOGGER.info("in order to reset the password for the kibana_system user.");
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("");
-        LOGGER.info("-----------------------------------------------------------------");
-        LOGGER.info("");
+    private void outputOnSuccess(SecureString elasticPassword, SecureString kibanaSystemPassword, PrintStream out) {
+        out.println();
```

Comment:
The stream could still have been closed, in which case we are not attached to a terminal. So we need to catch that case, and probably log a warning?

Reply:
Yes, I'll check this first before even generating the passwords. If we can't show them, there is no need to even generate them